beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -254,8 +254,8 @@ int network_utils::get_iface_info(network_utils::iface_info &info, const std::st
         }
     }
 
-    if (info.mac.empty()) {
-        network_utils::linux_iface_get_mac(iface_name, info.mac);
+    if (info.mac.empty() && !network_utils::linux_iface_get_mac(iface_name, info.mac)) {
```

Comment:
I personally don't like to depend on the short-circuiting of `&&` like this. It's a bit like `goto` IMO. It's acceptable to use it, but only for specific patterns. In this particular case, I think it's confusing because the condition of empty has no direct bearing on returning -1; the return -1 only applies to the `linux_iface_get_mac` call.

Ideally, I like to have a coding rules library with the "approved" patterns for the project.

An example of a short-circuit pattern that I would approve is the following.
```c
if (!foo() ||
    !bar() ||
    memcmp(x, y, sizeof(x))) {
        return -1;
}
```
i.e. a sequence of tests or operations with exit in case of error.

Actually, in C++ I'd even prefer to use exceptions instead of status code returns (in most cases). But since that is not at all the coding style used at the moment, let's stick to status code returns.

Bottom line: in this specific case, I'd write it like this:
```suggestion
    if (info.mac.empty()) {
        if(!network_utils::linux_iface_get_mac(iface_name, info.mac)) {
            return -1;
        }
    }
```

Reply:
Ok, changed. But using exceptions has its drawbacks and it was decided by the original beerocks architect not to use them.